### DISCUSSION TITLE
Lets use MyPy! (please vote)

### DISCUSSION THREAD

**Discussion by shroominic** [AnswerDotAI/fasthtml#562] [Polls]
MyPy is important because it adds static type checking to Python, allowing developers to catch type-related errors before runtime. By enforcing type annotations, MyPy makes code more predictable, self-documenting, and easier to understand, especially in large codebases or collaborative projects. This early error detection reduces debugging time, improves code reliability, and helps prevent subtle bugs that might otherwise go unnoticed. Additionally, MyPy can improve IDE support with better autocompletion and type hints, enhancing overall development efficiency.

**Comment by ZeeD** [comment]
FWIW I see that the code is already mildly annotated - and I also see a bunch of explicit `.pyi` files.
I would try to just start marking the library with a `py.typed` file in the package, so that the users of the library may have `mypy` work on it.
ATM if I run `mypy` on my project that uses fasthtml all I get is:

```
main.py:1: error: Skipping analyzing "fasthtml.components": module is installed, but missing library stubs or py.typed marker  [import-untyped]
main.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
main.py:3: error: Skipping analyzing "fasthtml.core": module is installed, but missing library stubs or py.typed marker  [import-untyped]
main.py:4: error: Skipping analyzing "fasthtml.fastapp": module is installed, but missing library stubs or py.typed marker  [import-untyped]
main.py:10: error: Untyped decorator makes function "get" untyped  [misc]
```

**Comment by shroominic** [comment]
yeah we need a `py.typed`
